apps/emqx_cluster_link/src/emqx_cluster_link_router_syncer.erl

```diff
@@ -0,0 +1,478 @@
+child_spec(syncer, Actor, Incarnation, TargetCluster) ->
+    %% Route syncer process.
+    %% Initially starts in a "closed" state. Actor decides when to open it, i.e.
+    %% when bootstrapping is done. Syncer crash means re-bootstrap is needed, so
+    %% we just restart the actor in this case.
+    SyncerRef = ?SYNCER_REF(TargetCluster),
+    ClientName = ?CLIENT_NAME(TargetCluster),
+    syncer_spec(syncer, Actor, Incarnation, SyncerRef, ClientName).
+
+actor_spec(ChildID, ActorRef, Actor, Incarnation, TargetCluster) ->
+    #{
+        id => ChildID,
+        start => {?MODULE, start_link_actor, [ActorRef, Actor, Incarnation, TargetCluster]},
+        restart => permanent,
+        type => worker
+    }.
+
+syncer_spec(ChildID, Actor, Incarnation, SyncerRef, ClientName) ->
+    #{
+        id => ChildID,
+        start => {?MODULE, start_link_syncer, [Actor, Incarnation, SyncerRef, ClientName]},
+        restart => permanent,
+        type => worker
+    }.
+
+%%
+
+-record(st, {
+    target :: binary(),
+    actor :: binary(),
+    incarnation :: non_neg_integer(),
+    client :: {pid(), reference()},
+    bootstrapped :: boolean(),
+    reconnect_timer :: reference(),
+    actor_init_req_id :: binary(),
+    actor_init_timer :: reference(),
+    remote_actor_info :: undefined | map()
+}).
+
+mk_state(TargetCluster, Actor, Incarnation) ->
+    #st{
+        target = TargetCluster,
+        actor = Actor,
+        incarnation = Incarnation,
+        bootstrapped = false
+    }.
+
+init_actor(State = #st{}) ->
+    _ = erlang:process_flag(trap_exit, true),
+    {ok, State, {continue, connect}}.
+
+handle_continue(connect, State) ->
+    {noreply, process_connect(State)}.
+
+handle_call(_Request, _From, State) ->
+    {reply, ignored, State}.
+
+handle_cast(_Request, State) ->
+    {noreply, State}.
+
+handle_info({'EXIT', ClientPid, Reason}, St = #st{client = ClientPid}) ->
+    {noreply, handle_client_down(Reason, St)};
+handle_info(
+    {publish, #{payload := Payload, properties := #{'Correlation-Data' := ReqId}}},
+    St = #st{actor_init_req_id = ReqId}
+) ->
+    {actor_init_ack, #{result := Res} = AckInfoMap} = emqx_cluster_link_mqtt:decode_resp(Payload),
+    St1 = St#st{
+        actor_init_req_id = undefined, actor_init_timer = undefined, remote_actor_info = AckInfoMap
+    },
+    case Res of
+        ok ->
+            {noreply, post_actor_init(St1)};
+        Error ->
+            ?SLOG(error, #{
+                msg => "failed_to_init_link",
+                reason => Error,
+                target_cluster => St#st.target,
+                actor => St#st.actor,
+                remote_link_proto_ver => maps:get(proto_ver, AckInfoMap, undefined)
+            }),
+            %% TODO: It doesn't fit permanent workers/one_for_all restart/strategy.
+            %% The actor may be kept alive with some error status instead (waiting for a user intervention to fix it)?
+            {stop, {shutdown, Error}, St1}
```

Comment:
Yeah I also think that handshake failure is essentially the same as connection failure from the operations perspective: both are usually caused by misconfiguration and thus expected to be transient. We'd likely want to behave similarly to bridges here: have a _link status_ somewhere and try to reestablish the communication channel from time to time.

Reply:
Added error reason and status to the actor state, added alarms for errors.